lib/iris/tests/unit/aux_factory/test_AtmosphereSigmaFactory.py

```diff
@@ -0,0 +1,285 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Unit tests for the
+`iris.aux_factory.AtmosphereSigmaFactory` class.
+
+"""
+
+from unittest import mock
+
+from cf_units import Unit
+import numpy as np
+
+from iris.aux_factory import AtmosphereSigmaFactory
+from iris.coords import AuxCoord, DimCoord
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+
+class Test___init__(tests.IrisTest):
+    def setUp(self):
+        self.pressure_at_top = mock.Mock(units=Unit("Pa"), nbounds=0, shape=())
+        self.sigma = mock.Mock(units=Unit("1"), nbounds=0)
+        self.surface_air_pressure = mock.Mock(units=Unit("Pa"), nbounds=0)
+        self.kwargs = dict(
+            pressure_at_top=self.pressure_at_top,
+            sigma=self.sigma,
+            surface_air_pressure=self.surface_air_pressure,
+        )
+
+    def test_insufficient_coordinates(self):
+        with self.assertRaises(ValueError):
+            AtmosphereSigmaFactory()
+        with self.assertRaises(ValueError):
+            AtmosphereSigmaFactory(
+                pressure_at_top=None,
+                sigma=self.sigma,
+                surface_air_pressure=self.surface_air_pressure,
+            )
+        with self.assertRaises(ValueError):
+            AtmosphereSigmaFactory(
+                pressure_at_top=self.pressure_at_top,
+                sigma=None,
+                surface_air_pressure=self.surface_air_pressure,
+            )
+        with self.assertRaises(ValueError):
+            AtmosphereSigmaFactory(
+                pressure_at_top=self.pressure_at_top,
+                sigma=self.sigma,
+                surface_air_pressure=None,
+            )
```

Comment:
Could you break down the tests here (and in the rest of the file) so that there is one call `AtmosphereSigmaFactory` (or whichever method it is testing) per test? For example, `test_insufficient_coordinates` would look like:

```
    def test_insufficient_coordinates_no_args(self):
        with self.assertRaises(ValueError):
            AtmosphereSigmaFactory()

    def test_insufficient_coordinates_no_ptop(self):
        with self.assertRaises(ValueError):
            AtmosphereSigmaFactory(
                pressure_at_top=None,
                sigma=self.sigma,
                surface_air_pressure=self.surface_air_pressure,
            )

     def test_insufficient_coordinates_no_sigma(self):
        with self.assertRaises(ValueError):
            AtmosphereSigmaFactory(
                pressure_at_top=self.pressure_at_top,
                sigma=None,
                surface_air_pressure=self.surface_air_pressure,
            )

    def test_insufficient_coordinates_no_ps(self):            
        with self.assertRaises(ValueError):
            AtmosphereSigmaFactory(
                pressure_at_top=self.pressure_at_top,
                sigma=self.sigma,
                surface_air_pressure=None,
            )
```

I see that in the unit tests for other aux coord factories we have a similar pattern to what you have done here, but that code was written a long time ago! In more recent times we have tried to aim for one call to the function per test. It does results in more boilerplate, and you have to come up for a name for each test, but it is more in-keeping with the idea of what a unit test should be. If a unit test includes multiple individual tests (e.g. having multiple calls to the function) in a single test then, if the first test fails, it won't run the other tests as it has encountered the error so you don't know whether they are working until you fix the first test.
Having a single test in each `def test...` ensures that they all run, and you get a complete of what is working in a single run of the tests.

So could you split out these test in this file so that each test function only has a single call to the function it is testing?